src/widgets/appointments/appointments-overview.component.tsx

```diff
@@ -37,36 +41,44 @@ export default function AppointmentsOverview(props: AppointmentOverviewProps) {
 
   return (
     <>
-      {patientAppointments && patientAppointments.length > 0 ? (
+      {patientAppointments?.length > 0 ? (
         <SummaryCard
-          name="Appointments"
+          name={t("Appointments")}
           link={appointmentsPath}
+          addComponent={AppointmentsForm}
           showComponent={() =>
-            openWorkspaceTab(AppointmentsForm, "Appointments Form")
+            openWorkspaceTab(AppointmentsForm, `${t("Appointments Form")}`)
           }
-          addComponent={AppointmentsForm}
         >
           <table
             className={`omrs-type-body-regular ${styles.appointmentTable}`}
           >
             <thead>
               <tr>
-                <td>Date</td>
-                <td>Service Type</td>
-                <td colSpan={2}>Status</td>
+                <td>
+                  <Trans i18nKey="date">Date</Trans>
+                </td>
+                <td>
+                  <Trans i18nKey="serviceType">Service Type</Trans>
+                </td>
+                <td colSpan={2}>
+                  <Trans i18nKey="status">Status</Trans>
+                </td>
               </tr>
             </thead>
             <tbody>
-              {patientAppointments.slice(0, 5).map(appointment => {
+              {patientAppointments?.slice(0, 5).map(appointment => {
                 return (
-                  <tr key={appointment.uuid}>
-                    <td test-id="startDate">
-                      {dayjs(appointment.startDateTime).format("DD-MMM-YYYY")}
+                  <tr key={appointment?.uuid}>
```

Comment:
Instead of using `appointment?.` everywhere in the map - use `filter` before the `slice` (i.e., only consider non-undefined entries), e.g.,

```ts
patientAppointments?.filter(m => !!m).slice(0, 5).map(appointment => ...)
```

Reply:
👌